packages/react-instantsearch/src/core/createInstantSearchManager.js

```diff
@@ -29,6 +29,9 @@ export default function createInstantSearchManager({
   helper.on('result', handleSearchSuccess);
   helper.on('error', handleSearchError);
 
+  let derivedHelpers = {};
+  let indexMapping = {}; // keep track of the original index where the parameters applied
```

Comment:
Maybe the comment and initial values are not exactly the right one because we expect indexMapping to always be manipulated in a way that we "keep track of something between search calls" while it's always being re-assigned to an empty object at every search call.